haddock-api/src/Haddock/Interface.hs

```diff
@@ -123,11 +124,12 @@ processModules verbosity modules flags extIfaces = do
   out verbosity verbose "Renaming interfaces..."
   let warnings = Flag_NoWarnings `notElem` flags
   dflags <- getDynFlags
-  interfaces'' <- withTimingM "renameAllInterfaces" (const ()) $ do
-    let (interfaces'', msgs) =
-           runErrMsgM $ mapM (renameInterface dflags (ignoredSymbols flags) links warnings) interfaces'
-    liftIO $ mapM_ (BSL.putStrLn . toLazyByteString) (errorMessagesToList msgs)
-    pure interfaces''
+  let !ignoredSymbolSet = Set.fromList (ignoredSymbols flags)
+  interfaces'' <-
+    withTimingM "renameAllInterfaces" (const ()) $
+      for interfaces' $ \i -> do
+        withTimingM ("renaming interface: " <+> pprModuleName (moduleName (ifaceMod i))) (const ()) $
+          renameInterface dflags ignoredSymbolSet links warnings i
```

Comment:
The rewrite and timing info here is from a previous incarnation, where I was doing something like:

```haskell
for interfaces' $ \i -> 
  (i', msgs) <- runErrMsgM $ renameInterface dflags ignoredSymbols links warnings i
  mapM_ (BSL.putStrLn . toLazyByteString) (errorMessagesToList msgs)
  pure i'
```

However, a *further* refactor dropped the `ErrMsgM` entirely in favor of making the function polymorphic over `ReportErrorMessage`. An instance is defined for `Ghc` that prints each item out as it comes.